data-runtime/src/main/java/io/micronaut/data/runtime/intercept/criteria/AbstractSpecificationInterceptor.java

```diff
@@ -139,9 +142,11 @@ protected final <E, QR> PreparedQuery<E, QR> preparedQueryForCriteria(Repository
                 }
         );
 
+        Set<JoinPath> joinPaths = methodsJoinPaths.computeIfAbsent(methodKey, repositoryMethodKey -> AssociationUtils.getJoinFetchPaths(context, false));
```

Comment:
Move this to the if block where it's used, so it's not always run.

Reply:
Makes sense, thanks. Done